sqs/src/main/scala/akka/stream/alpakka/sqs/SqsSourceStage.scala

```diff
@@ -19,20 +19,69 @@ object SqsSourceSettings {
   def create(waitTimeSeconds: Int, maxBufferSize: Int, maxBatchSize: Int): SqsSourceSettings =
     SqsSourceSettings(waitTimeSeconds, maxBufferSize, maxBatchSize)
 
+  def create(waitTimeSeconds: Int,
+             maxBufferSize: Int,
+             maxBatchSize: Int,
+             attributeNames: util.List[String],
+             messageAttributeNames: util.List[String]): SqsSourceSettings =
+    SqsSourceSettings(waitTimeSeconds,
+                      maxBufferSize,
+                      maxBatchSize,
+                      attributeNames.asScala,
+                      messageAttributeNames.asScala)
+
 }
 
 //#SqsSourceSettings
 final case class SqsSourceSettings(
     waitTimeSeconds: Int,
     maxBufferSize: Int,
-    maxBatchSize: Int
+    maxBatchSize: Int,
+    attributeNames: Seq[String] = Seq(),
+    messageAttributeNames: Seq[String] = Seq()
 ) {
   require(maxBatchSize <= maxBufferSize, "maxBatchSize must be lower or equal than maxBufferSize")
   // SQS requirements
   require(0 <= waitTimeSeconds && waitTimeSeconds <= 20,
           s"Invalid value ($waitTimeSeconds) for waitTimeSeconds. Requirement: 0 <= waitTimeSeconds <= 20 ")
   require(1 <= maxBatchSize && maxBatchSize <= 10,
           s"Invalid value ($maxBatchSize) for maxBatchSize. Requirement: 1 <= maxBatchSize <= 10 ")
+
+  private val validNames = Seq(
+    "All",
+    "Policy",
+    "VisibilityTimeout",
+    "MaximumMessageSize",
+    "MessageRetentionPeriod",
+    "ApproximateNumberOfMessages",
+    "ApproximateNumberOfMessagesNotVisible",
+    "CreatedTimestamp",
+    "LastModifiedTimestamp",
+    "QueueArn",
+    "ApproximateNumberOfMessagesDelayed",
+    "DelaySeconds",
+    "ReceiveMessageWaitTimeSeconds",
+    "RedrivePolicy",
+    "FifoQueue",
+    "ContentBasedDeduplication",
+    "KmsMasterKeyId",
+    "KmsDataKeyReusePeriodSeconds"
+  )
+
+  require(attributeNames.forall(name => validNames.contains(name)), s"AttributeName must be one of $validNames")
+
+  require(
+    messageAttributeNames.forall(name => name.matches("[0-9a-zA-Z_\\-.]+")),
+    "MessageAttributeNames may only contain alphanumeric characters and the underscore (_), hyphen (-), and period (.)"
+  )
+
+  require(
```

Comment:
This could also be done with a type instead to fail "faster", do the require directly in the constructor and know that it is safe here. Not sure if it is the other names plus custom names? Could either be a sealed trait which `AttributeName` extends plus an extra `CustomAttributeName` subtype or just single type which always takes that string.